ddtrace/internal/telemetry/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""
+Instrumentation Telemetry API.
+This is normally started automatically by ``ddtrace-run`` when the
+``DD_INSTRUMENTATION_TELEMETRY_ENABLED`` variable is set.
+To start the service manually, invoke the ``enable`` method::
+    from ddtrace.internal import telemetry
+    telemetry.enable()
+"""
+
+from ...internal import forksafe
+from ..service import ServiceStatus
+from .writer import TelemetryWriter
+
+
+TELEMETRY_WRITER = TelemetryWriter()
```

Comment:
I thought the goal was to avoid the TelemetryWriter class from self-managing the singleton?

I am 👍🏻 on `_TELEMETRY_WRITER`

The benefit of taking the current approach is if (when) we move to shared library for the client, we can keep this "public" interface the same, the actual buffering/flushing of events can stay a mystery.

Reply:
> I thought the goal was to avoid the TelemetryWriter class from self-managing the singleton?

The difference is that there just be special instance, like `ddtrace.tracer`... so no class methods/attributes or other global state.

The public interface just becomes the TelemetryWriter public interface on a class level vs a module level one.

Reply:
oh got it, just doing `from ddtrace.internal.telemetry import telemetry_writer` like we have with the tracer?

Reply:
I don't think I have a strong opinion here, I am ok with either approach, probably achieves the same, just whether we have extra layer of indirection or not

Reply:
right yes, I don't feel strongly either since this is an internal API and easy to change. Just felt like a bit of a smell to proxy everything and the thread unsafe methods using class/global state, etc

Reply:
I had not considered the thread safety of those methods, which definitely complicates things since the `TelemetryWriter` class should own the lock.

Do you have a proposed structure for how this should be setup `enable()`/`reset()`/`disable()`?

Have them be normal instance methods and they just manage the parent periodic service/thread?

Before it was managing a singleton, but your proposal is to have it just manage itself/the single instance right?

Reply:
Yeah my proposed API is that `enable()`/`reset()`/`disable()` encapsulated in `TelemetryWriter` as instance methods/attributes.


> Before it was managing a singleton, but your proposal is to have it just manage itself/the single instance right?

👍 

just seems simpler at the moment 🤷‍♂️ again it's just an internal API so doesn't matter much

Reply:
Well, the part that matters is the thread safety issue you called out, I think if we need to use a lock here we either need to introduce a new module level lock for the existing API, or need to move to the instance methods like you are proposing so we can use the instance lock on `TelemetryWriter`.

@mabdinur what do you think?

Reply:
I think we should encapsulate the enable()/reset()/disable() functions in the writer and use the instance lock for thread safety. I will also remove the `telemetry.add_integration()` method since it just calls one method on the writer, we can use `TelemetryWriter.add_integration` instead.

Reply:
Will use `TELEMETRY_WRITER.add_integrations()` in _monkey.py instead of `telemetry.add_integration() `

Reply:
```suggestion
telemetry_writer = TelemetryWriter()
```